### PR TITLE
ref(notifications): Refactor notification utilities for msteams

### PR DESCRIPTION
**Pull request by 97varun**
Move notification utility functions to a common file for reuse in msteams notifications.

* Make a slight modification to the identity query to exclude msteams identity rows which have a `team_id` as `external_id` instead of a `user_id`.

### CODE REVIEW

File: src/sentry/integrations/notifications.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    identities = (
+        Identity.objects.filter(
+            idp__type=EXTERNAL_PROVIDERS[provider],
+            user=user.id,
+        )
+        # For Microsoft Teams integration, initially we create rows in the
+        # identity table with the external_id as a team_id instead of the user_id.
+        # We need to exclude rows where this is NOT updated to the user_id later.
+        .exclude(external_id=F("idp__external_id")).select_related("idp")
```
Comment (97varun) [inline src/sentry/integrations/notifications.py:41]: Rows where `identity.external_id == idp.external_id` will be excluded.

File: src/sentry/integrations/notifications.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            )
+            .select_related("integration")
+            .get()
+        )
+    except ExternalActor.DoesNotExist:
+        return {}
+    return {external_actor.external_id: external_actor.integration}
+
+
+def get_integrations_by_channel_by_recipient(
```
Comment (AniketDas-Tekky) [inline src/sentry/integrations/notifications.py:87]: Are the tests updated to call this instead?

Comment (97varun) [reply src/sentry/integrations/notifications.py:87]: @AniketDas-Tekky, thanks for pointing out! This function wasn't being tested directly. I added some tests which tests the new functionality as well.

File: tests/sentry/integrations/test_notification_utilities.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class TestNotificationUtilities(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.notification = DummyNotification(self.organization)
+
+        self.integration = install_slack(self.organization)
+
+        self.idp = IdentityProvider.objects.create(type="slack", external_id="TXXXXXXX1", config={})
```
Comment (AniketDas-Tekky) [inline tests/sentry/integrations/test_notification_utilities.py:16]: we have fixtures to help with setup. In this case `self.create_identity_provider`

https://github.com/getsentry/sentry/blob/master/src/sentry/testutils/fixtures.py#L391

You can do the same with identity

File: tests/sentry/integrations/test_notification_utilities.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        self.identity = Identity.objects.create(
+            external_id="UXXXXXXX1",
```
Comment (AniketDas-Tekky) [inline tests/sentry/integrations/test_notification_utilities.py:18]: nit: save strings as test constants
```
self.external_id_1 = "UXXXXXXX1"
# ...
    external_id=self.external_id_1
```

File: tests/sentry/integrations/test_notification_utilities.py (hunk 3 above)
Comment (97varun) [reply tests/sentry/integrations/test_notification_utilities.py]: This is so good, reduced a lot of boilerplate.

File: tests/sentry/integrations/test_notification_utilities.py (hunk 4 above)
Comment (97varun) [reply tests/sentry/integrations/test_notification_utilities.py]: Updated.

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request merged**